microsoft-edge/extensions-chromium/developer-guide/sidebar.md

```diff
@@ -0,0 +1,625 @@
+
+Promises are supported in Manifest V3 and later, but callbacks are provided for backward compatibility.  You cannot use a promise along with a callback on the same function call.  The promise resolves as the same type that's passed to the callback.
+
+
+<!-- ====================================================================== -->
+#### getPanelBehavior
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#method-getPanelBehavior -->
+
+Returns the extension's current sidebar behavior.
+
+
+<!-- ---------- -->
+**SYNTAX**
+
+```js
+chrome.sidePanel.getPanelBehavior(
+  callback?: function,
+)
+```
+
+* Promise - This method an return its result via a Promise, in Manifest V3 or later.
+* Pending - This method is coming soon and is not yet in a stable release of Microsoft Edge.
+
+
+<!-- ---------- -->
+**PARAMETERS**
+
+
+###### callback
+
+Type: {} function, optional
+
+The `callback` parameter looks like:
+
+```js
+(behavior: PanelBehavior) => void
+```
+
+Parameter type: [PanelBehavior](#panelbehavior)
+
+
+<!-- ---------- -->
+**RETURNS**
+
+Promise<[PanelBehavior](#panelbehavior)>
+
+Promises are supported in Manifest V3 and later, but callbacks are provided for backward compatibility.  You cannot use a promise together with a callback on the same function call.  The promise resolves as the same type that's passed to the callback.
+
+
+<!-- ====================================================================== -->
+#### setOptions
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#method-setOptions -->
+
+Configures the sidebar.
+
+
+<!-- ---------- -->
+**SYNTAX**
+
+```js
+chrome.sidePanel.setOptions(
+  options: PanelOptions,
+  callback?: function,
+)
+```
+
+* Promise - This method an return its result via a Promise, in Manifest V3 or later.
+
+
+<!-- ---------- -->
+**PARAMETERS**
+
+###### options
+
+Type: [PanelOptions](#paneloptions)
+
+
+###### callback
+
+Type: {} function, optional
+
+The `callback` parameter looks like:
+
+```js
+() => void
+```
+
+
+<!-- ---------- -->
+**RETURNS**
+
+Promise<`Void`>
+
+Promises are supported in Manifest V3 and later, but callbacks are provided for backward compatibility. You cannot use both on the same function call. The promise resolves with the same type that is passed to the callback.
+
+
+<!-- ====================================================================== -->
+#### setPanelBehavior
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#method-setPanelBehavior -->
+
+Configures the extension's sidebar behavior.  This is an upsert (update or insert) operation.
+
+
+<!-- ---------- -->
+**SYNTAX**
+
+```js
+chrome.sidePanel.setPanelBehavior(
+  behavior: PanelBehavior,
+  callback?: function,
+)
+```
+
+* Promise - This method an return its result via a Promise, in Manifest V3 or later.
+* Pending - This method is coming soon and is not yet in a stable release of Microsoft Edge.
```

Comment:
Why are we saying this only here?

Reply:
todo:
see each section such as https://developer.chrome.com/docs/extensions/reference/sidePanel/#method-setPanelBehavior and see where "Promise" label appears and where "Pending" label appears.  Make sure consistently matches labels in each section.

Reply:
@abhishek-shanthkumar - are you thinking of adding a statement in released-features or roadmap.md?
https://learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/whats-new/released-features
https://learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/whats-new/roadmap

Reply:
Anything coming up should be on the roadmap

Reply:
Added statement to Roadmap & top of article